lib/matplotlib/testing/conftest.py

```diff
@@ -76,3 +76,26 @@ def mpl_image_comparison_parameters(request, extension):
         yield
     finally:
         delattr(func.__wrapped__, 'parameters')
+
+
+@pytest.fixture
+def pd(request):
+    '''fixture to import and configure pandas'''
+
+    pd = pytest.importorskip('pandas')
+    if pd:
+        try:
+            from pandas.plotting import (
+                register_matplotlib_converters as register)
+        except ImportError:
+            from pandas.tseries.converter import register
+        register()
+
+        try:
+            from pandas.plotting import (
+                deregister_matplotlib_converters as deregister)
+            request.addfinalizer(deregister)
```

Comment:
a yield fixture (in the style of a context manager) seems mildly more pythonic (https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code).

Reply:
@tacaswell let us know if you want to make the change or I can make it in a separate pr too

Reply:
@anntzer sorry, been traveling, feel free to push to this branch.